src/discretization/regular.jl

```diff
@@ -23,12 +23,20 @@ function discretize(geometry::Geometry, method::RegularDiscretization)
     tmesh = appendtopo(geometry, tgrid)
     SimpleMesh(collect(verts), tmesh)
   else
-    box = boundingbox(geometry)
-    grid = discretize(box, method)
-    view(grid, geometry)
+    discretizewithinbox(geometry, method)
   end
 end
 
+discretize(triangle::Triangle, method::RegularDiscretization) = discretizewithinbox(triangle, method)
```

Comment:
Maybe add a comment why we need to overload this method for `Triangle` and `Tetrahedron`? IIUC, it's because the parametric functions for them are defined for barycentric coordinates in contrast to $[0, 1]^n$. I needed a bit to realize that.

Reply:
Thank you for the suggestion. I will add this comment 👍🏽